internal/core/src/clustering/analyze_c.cpp

```diff
@@ -208,22 +209,34 @@ AppendSegmentNumRows(CAnalyzeInfo c_analyze_info,
 }
 
 CStatus
-SerializeAnalyzeAndUpLoad(CAnalyze analyze, CBinarySet* c_binary_set) {
+GetAnalyzeResultMeta(CAnalyze analyze,
+                     char* centroid_path,
+                     int64_t* centroid_file_size,
+                     void* id_mapping_paths,
+                     int64_t* id_mapping_sizes) {
     auto status = CStatus();
     try {
         AssertInfo(analyze,
                    "failed to serialize analyze to binary set, passed index "
                    "was null");
         auto real_analyze =
-            reinterpret_cast<milvus::indexbuilder::MajorCompaction*>(analyze);
-        auto binary =
-            std::make_unique<knowhere::BinarySet>(real_analyze->Upload());
-        *c_binary_set = binary.release();
+            reinterpret_cast<milvus::clustering::Clustering*>(analyze);
```

Comment:
If `GetClusteringResultMeta` is not template related, feel free to:
1. Put template on the function level
2. Give a static function wrapper